packages/dialog/theme/lumo/vaadin-dialog-styles.js

```diff
@@ -23,39 +23,81 @@ const dialogOverlay = css`
     -webkit-tap-highlight-color: transparent;
   }
 
+  [part='overlay']:focus-visible {
+    box-shadow: 0 0 0 2px var(--lumo-primary-color-50pct), 0 0 0 1px var(--lumo-shade-5pct), var(--lumo-box-shadow-xl);
+  }
```

Comment:
Let's split this change out to a separate PR as it's not directly header / footer related.

Right now there is also an inconsistency in terms of setting this outline:

1. When you use `<vaadin-dialog opened>` the outline is shown as dialog opens,
2. When opening dialog in the dev page on button click, the outline is not shown,
3. But when pressing <kbd>Enter</kbd> on the button, dialog outline is shown.